### PR TITLE
Upgrade to PHP7

### PR DESCRIPTION
**Pull request by mrtcode**
php-memcached 3 is needed.

### CODE REVIEW

File: controllers/ApiController.php [hunk 1]
```diff
@@ -813,7 +813,7 @@ protected function checkSingleObjectWriteVersion($objectType, $obj=null, $json=f
 	 */
 	protected function checkLibraryIfUnmodifiedSinceVersion($required=false) {
 		if (Z_CONFIG::$TESTING_SITE && !empty($_GET['skipetag'])) {
```
Comment (mrtcode) [inline controllers/ApiController.php]: What was the actual purpose of this 'if' ?

**Comment by mrtcode** [comment]
It's possible that there are more problems. Not all code was test tested. E.g. classic-sync logics. Actually any code that wasn't executed can potentially have incompatibilities with php7.

File: controllers/ApiController.php (hunk 1 above)
Comment (dstillman) [inline controllers/ApiController.php]: Just to make manual testing of successive writes with curl a little easier, without updating the version each time. I don't think I've used it in years, but for now can just change the `continue` to `return true`.

**Comment by dstillman** [comment]
I'm trying this with PHP 5.5.38 first and getting this error:

> Fatal error: Class 'Psr\Log\NullLogger' not found in /var/www/dataserver-dev/include/Elastica/lib/Elastica/Client.php on line 93

Is the updated Elastica necessary for PHP 7 support? If so, do we need to add something to the Composer config to get the latest version working?

**Comment by mrtcode** [comment]
I haven't tested the old Elastica version, because it also needs old Elasticsearch. I can do the tests if needed. Also the new Elastica version is probably not going to work with that old Elasticsearch. Should I test it? You will probably need to update Zend framework too.


New Elastica needs:
composer require psr/log:dev-master
composer require elasticsearch/elasticsearch

Any required configuration can be found:
https://github.com/mrtcode/zotero-server/blob/master/containers/dataserver/Dockerfile

**Pull request merged**